src/sql/azdata.proposed.d.ts

```diff
@@ -2073,6 +2073,382 @@ declare module 'azdata' {
 		export function openServerFileBrowserDialog(connectionUri: string, targetPath: string, fileFilters: FileFilters[]): Thenable<string | undefined>;
 	}
 
+	//#region Chart component model types
+
+	export type ChartType = 'bar' | 'bubble' | 'doughnut' | 'horizontalBar' | 'line' | 'pie' | 'polarArea' | 'radar' | 'scatter';
+
+	export interface ModelBuilder {
+		chart<TChartType extends ChartType, TData extends ChartData<TChartType>, TOptions extends ChartOptions<TChartType>>(): ComponentBuilder<ChartComponent<TChartType, TData, TOptions>, ChartComponentProperties<TChartType, TData, TOptions>>;
+	}
+
+	export interface ChartComponent<TChartType extends ChartType, TData extends ChartData<TChartType>, TOptions extends ChartOptions<TChartType>> extends Component, ChartComponentProperties<TChartType, TData, TOptions> {
+		onDidClick: vscode.Event<ChartClickEvent>;
+	}
+
+	export type ChartClickEvent = { label: string };
+
+	export interface ChartComponentProperties<TChartType extends ChartType, TData extends ChartData<TChartType>, TOptions extends ChartOptions<TChartType>> extends ComponentProperties {
+		/**
+		 * Unique ID for chart, used as the canvasID to render.
+		 */
+		chartId: string;
+
+		/**
+		 * Type of chart to build
+		 * @see ChartType for a list of supported chart types
+		 */
+		chartType: ChartType;
```

Comment:
Is the user supposed to be able to switch chart types on the fly? Or is the expectation that they'll just create a whole new chart at that point? If the latter, could this by `TChartType` then to guarantee it's the same as the type used for the generic typing?

Reply:
Ah yeah, I meant to make this change; fixed now.

The user cannot change chart types on the fly; the compatibility graph would be pretty sparse.